modules/flowable-task-service/src/main/java/org/flowable/task/service/impl/HistoricTaskInstanceQueryImpl.java

```diff
@@ -650,6 +652,34 @@ public HistoricTaskInstanceQuery taskName(String taskName) {
         return this;
     }
 
+    /**
+     * Only select tasks with a task id that is in the given list
+     *
+     * @param taskIdList
+     * @throws FlowableIllegalArgumentException When task id list is empty or <code>null</code> or contains <code>null String</code>.
+     */
+    @Override
+    public HistoricTaskInstanceQuery taskIdIn(Collection<String> taskIdList) {
+        if (taskIdList == null) {
+            throw new FlowableIllegalArgumentException("Task id list is null");
+        }
+        if (taskIdList.isEmpty()) {
+            throw new FlowableIllegalArgumentException("Task id list is empty");
+        }
+        for (String taskId : taskIdList) {
+            if (taskId == null) {
+                throw new FlowableIllegalArgumentException("None of the given task ids can be null");
+            }
+        }
```

Comment:
I don't think that we need to do this, we are not doing it for any other collection